src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs

```diff
@@ -94,11 +97,47 @@ private void EnsureOptions()
                 var dependencyContextOptions = GetDependencyContextCompilationOptions();
                 _parseOptions = GetParseOptions(_hostingEnvironment, dependencyContextOptions);
                 _compilationOptions = GetCompilationOptions(_hostingEnvironment, dependencyContextOptions);
+                _emitOptions = GetEmitOptions(dependencyContextOptions);
 
                 _optionsInitialized = true;
             }
         }
 
+        private EmitOptions GetEmitOptions(DependencyContextCompilationOptions dependencyContextOptions)
+        {
+            DebugInformationFormat debugInformationFormat;
+            if (string.IsNullOrEmpty(dependencyContextOptions.DebugType))
+            {
+                debugInformationFormat = SymbolsUtility.SupportsFullPdbGeneration() ?
+                    DebugInformationFormat.Pdb :
+                    DebugInformationFormat.PortablePdb;
+            }
+            else
+            {
+                // Based on https://github.com/dotnet/roslyn/blob/1d28ff9ba248b332de3c84d23194a1d7bde07e4d/src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs#L624-L640
```

Comment:
Can you please log an issue for them to expose a public parse method for this (or other guidance so we don't have to do this.

Reply:
https://github.com/dotnet/roslyn/issues/23997